runtime/compiler/env/J9CPU.hpp

```diff
@@ -45,12 +45,43 @@ class OMR_EXTENSIBLE CPU : public OMR::CPUConnector
    CPU() : OMR::CPUConnector() {}
    CPU(const OMRProcessorDesc& processorDescription) : OMR::CPUConnector(processorDescription) {}
 
+   /**
+    * @brief Contain the list of processor features utlizied by the compiler, initialized via TR::CPU::initializeFeatureMasks()
+    */
+   static OMRProcessorDesc _supportedFeatureMasks;
+
+   /**
+    * @brief _isSupportedFeatureMasksEnabled tells you whether _supportedFeatureMasks was used for masking out unused processor features
+    */
+   static bool _isSupportedFeatureMasksEnabled;
+
 public:
 
+   /** 
+    * @brief Returns the processor type and features that will be used by JIT and AOT compilations
+    * @param[in] omrPortLib : the port library
+    * @return TR::CPU
+    */
+   static TR::CPU detect(OMRPortLibrary * const omrPortLib);
+
+   /** 
+    * @brief Disable processor features based on input processor description, user options and whether they are utilized by the compiler
```

Comment:
I would replace "Disable" with "Sets".
Do we actually look at user options? I thought this was done in another part of the code which you showed it to me.

Reply:
We do not look at user options in "TR::CPU::customize()". And I agree that we should replace "disable" with "set".

processor description contains two piece of information 1. processor type 2. processor features.
The only case that we disable processor features is on Z, when you downgrade your processor from say z14 to z10, there are a couple of features you need to disable. `TR::CPU::customize()` will automatically do it for you. a z10 cpu type with z14 feature set won't make sense to the compiler.
That's the only case.

Reply:
Description on these two has been changed to:
```cpp 
  /** 
    * @brief A factory method used to construct an CPU object based on the underlying hardware
    * @param[in] omrPortLib : the port library
    * @return TR::CPU
    */
   static TR::CPU detect(OMRPortLibrary * const omrPortLib);

   /** 
    * @brief A factory method used to construct an CPU object based on user customized processorDescription
    * @param[in] OMRProcessorDesc : the processor description
    * @return TR::CPU
    */
   static TR::CPU customize(OMRProcessorDesc processorDescription);
```